hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -508,11 +508,16 @@ class HoodieSparkSqlWriterInternal {
               throw new UnsupportedOperationException(s"${writeConfig.getRecordMerger.getClass.getName} only support parquet log.")
             }
             instantTime = client.startCommit(commitActionType)
+            // if table has undergone upgrade, we need to reload table config
+            tableConfig = HoodieTableMetaClient.builder()
```

Comment:
Can we use `tableMetaClient.reloadTableConfig()` for this instead?

Reply:
yes, sure. 
Will follow this
```
            tableMetaClient.reloadTableConfig()
            tableConfig = tableMetaClient.getTableConfig
```